pytorch_lightning/callbacks/lambda_cb.py

```diff
@@ -0,0 +1,45 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Lambda Callback
+^^^^^^^^^^^^^^^
+
+Create a simple callback on the fly.
+
+"""
+
+from pytorch_lightning.callbacks.base import Callback
+
+
+class LambdaCallback(Callback):
+    r"""
+    Create a simple callback on the fly.
+
+    Args:
+        **kwargs: event listener supported by ``Callback``
+
+    Example::
+
+        >>> from pytorch_lightning import Trainer
+        >>> from pytorch_lightning.callbacks import LambdaCallback
+        >>> trainer = Trainer(callbacks=[LambdaCallback(setup=lambda *args: print('setup'))])
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
Can we make this more explicit? I.e. name every hook we support here? Otherwise people may not know what is supported and autocomplete won't be of any help here.

I know, that this increases the maintenance effort, but I think that this is essential for intuitive usage.

Reply:
You are right
If this callback supports auto-completion, it will cost more to maintain.  tensorflow supports autocomplete, but they have far fewer hooks than we do.  I'm curious about the opinions of other maintainers.

Reply:
I agree with Justus here. We can write a test that will fail if someone adds a new callback method and does not update the LambdaCallback init (or renames, or deleteds callback methods).